src/bfs_tree.h

```diff
@@ -30,22 +30,29 @@
 typedef struct bfs_node *BFSTree;
 typedef struct bfs_node  BFSNode;
 
-#define EMPTY_TREE (BFSTree *)0
-#define EMPTY_NODE (BFSNode *)0
+#define EMPTY_TREE (void *)0
```

Comment:
Explain me the reason behind this change. I don't see any reason to change it, as it is a null `BFSTree` node instead of a `void` one.

Reply:
I did this because I thought it would be easier to understand. For example, in some functions that return BFSTree type, we sometimes check if a node is null instead of the tree itself. Like here:

```c
BFSTree *bfs_tree_insert(BFSTree *root, char path[BFS_PATH_SIZE]) {
    if(check_pointer(root)) {
        return EMPTY_TREE;
    }
    else if(check_pointer(*root)) {
        return EMPTY_TREE;
    }
    ...
}
```
Without this change, we can't return EMPTY_NODE in a BFSTree type function or EMPTY_TREE in a BFSNode type function, like this:

```c
BFSTree *bfs_tree_insert(BFSTree *root, char path[BFS_PATH_SIZE]) {
    if(check_pointer(root)) {
        return EMPTY_TREE;
    }
    else if(check_pointer(*root)) {
        return EMPTY_NODE;
    }
    ...
}
```

I know it's not very important, but I also think it does not make much sense to return EMPTY_TREE instead of EMPTY_NODE when we're checking a node instead of the whole tree or the opposite. But I understand that it is not quit right to do it this way, so I am returning to the old pattern.